optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverJob.java

```diff
@@ -206,6 +215,12 @@ public Duration getSolvingDuration() {
         return Duration.ofMillis(endingSystemTimeMillis - startingSystemTimeMillis);
     }
 
+    void close() {
+        if (consumerSupport != null) {
```

Comment:
If we're never setting this to null, why not make `consumerSupport` final?

Reply:
I will set it to null. I don't want to make it final as that would require creating the instance in the constructor. As a result, it would create a new thread pool when the `SolverJob` is created; now the thread pool is created when the `SolverJob` starts solving.